spec/services/hyrax/collections/nested_collection_query_service_spec.rb

```diff
@@ -217,7 +176,7 @@
           it 'returns an array of collections of the same collection type excluding the given collection' do
             expect(scope).to receive(:can?).with(:read, coll_c).and_return(true)
             expect(described_class).to receive(:query_solr).with(collection: coll_c, access: :deposit, scope: scope, limit_to_id: nil, nest_direction: :as_parent).and_call_original
-            expect(subject.map(&:id)).to contain_exactly(coll_a.id, another.id)
+            expect(subject.map(&:id)).to contain_exactly(another.id)
```

Comment:
I think this test was incorrect before. If A -> B -> C -> D -> E then I wouldn't expect A to be valid as an "available parent" for C because it already is. I believe this is actually prevented by the validator if you attempt to do this in the UI.

Reply:
I believe in the current implementation, C can have A as an ancestor through B and as a direct parent. So

A -> B -> C -> D -> E
A -> C -> D -> E

at the same time. BUT I do think that works more by accident than by design.

Reply:
Allowing E to be a direct child of C and indirect child of C was intentional but perhaps from a theoretical stand point instead of anything concrete.

Reply:
FWIW I tested again locally and this works for User Collections, but when I attempt to do this with a collection type that doesn't allow multiple membership, I get an error:
```
Hyrax::SingleMembershipError in Hyrax::Dashboard::NestCollectionsController#create_relationship_within
Error: You have specified more than one of the same single-membership collection type (type: Nesting only, collections: C and D)
```

Reply:
I'll look into restoring this functionality.

Reply:
I added support for this back in with https://github.com/samvera/hyrax/pull/5850/commits/dfc2f0963dbb45d77998954b761a15103969c384